### ISSUE TITLE
Bicycle route text shield are hard to read

### ISSUE THREAD

**Issue by Florimondable** [cyclosm/cyclosm-cartocss-style#404]
« Perhaps it would be more correct to file a bug report / feature request on github, but as it still seems early days (v. 0.3.6) in CyclOSM, as well as I'm an old hand in OSM with a wiki account (and I read and write a lot of it), so I'm comfortable asking this here. I find the route "shields" of dark blue (lcn) and purple (rcn) with white alphanumeric text on them to be exceedingly hard to read. Even as I use a 5K Retina screen, I don't think it is my video or my eyeballs, I think it is the choice of text color, and likely font, type size and most probably "weight" of the font used which contributes to this. (I'm an ex-employee of both Apple and Adobe, though I do have eyeballs. Plus, I like this renderer VERY much and want to find its many wonderful features actually usable). Is there any way that on lcn and rcn route shields, a "heavier weight" font (bold?, extra bold?) can be used to make these alphanumerics easier to read? Thank you. Stevea (talk) 19:52, 7 July 2020 (UTC) »
https://wiki.openstreetmap.org/wiki/Talk:CyclOSM

**Comment by michaelblyons** [comment]
Is the route shield width a constant &times; number-of-chars? Having a `W` character (relatively wide) in the route shield pushes the edges of the first and last char off the shield.

**Comment by Florimondable** [comment]
@michaelblyons do you have an example ?

**Comment by michaelblyons** [comment]
<img width="30" alt="Screen Shot 2020-12-22 at 11 05 43 AM" src="https://user-images.githubusercontent.com/2701605/102908593-9c57f200-446f-11eb-831c-0ff3eabd1199.png"> shown [here](https://www.cyclosm.org/#map=13/39.0584/-77.0385/cyclosm).

**Comment by Phyks** [comment]
Shields are statically generated in advance, for a fixed number of chars. This is done in this script: https://github.com/cyclosm/cyclosm-cartocss-style/blob/master/scripts/generate_shields.py.

We used to have shields with extra margins which were making them too prominent in dense areas (https://www.cyclosm.org/#map=13/48.8633/2.3463/cyclosm for example). It might be worth doing some try and error with this script to improve the margins for wide characters.

If you are interested in this topic, let me know and I'll be happy to walk you through this.

Best,

**Comment by michaelblyons** [comment]
Thanks for the notes! I'll take a look at the Python and at your densely-routed area.

**Comment by hiddewie** [comment]
The last months the text on the shields is very 'thin' and the letters are almost impossible to read for some routes. If this is another issue I will open a separate one.

See screenshot at https://www.cyclosm.org/#map=9/37.5282/-7.2839/cyclosm


![image](https://user-images.githubusercontent.com/1073881/112680616-e1743180-8e6d-11eb-86fb-5e08a7fd8881.png)

Other screenshot at https://www.cyclosm.org/#map=8/43.967/0.379/cyclosm

![image](https://user-images.githubusercontent.com/1073881/112680653-f18c1100-8e6d-11eb-9f9d-0629d70bde64.png)

**Comment by Phyks** [comment]
Could you date "the last months"?

Nothing should have changed: https://github.com/cyclosm/cyclosm-cartocss-style/blame/master/labels.mss#L587

The only possible change seems to be https://github.com/cyclosm/cyclosm-cartocss-style/commit/f4de5270f001413c106272aa4abc04555c6e7348 which is more than a year old.

**Comment by Florimondable** [comment]
The shields in your pictures @hiddewie are indeed hard to read.
But on my side the rendering is ok :
![Screenshot_20210327_165733](https://user-images.githubusercontent.com/47089717/112726412-9dd90080-8f1d-11eb-909a-ea853bde4f00.png)
![Screenshot_20210327_165719](https://user-images.githubusercontent.com/47089717/112726414-9e719700-8f1d-11eb-82da-3be0e0606229.png)

May be just an issue with the compression of your pictures ?

**Comment by hiddewie** [comment]
> Could you date "the last months"?

Not really. I have been seeing it for quite some time but I never made an issue in this repository, sorry!

I will locally try a `git bisect` and try to reproduce the problem.

> But on my side the rendering is ok

Oh that is very interesting. The screenshots you just posted show exactly the same problems in my browser: the labels are very hard to read.

Then I compared to Firefox. Comparison screenshot below (left: Brave browser (Chromium based), right: Firefox browser):

![image](https://user-images.githubusercontent.com/1073881/112761286-e1ee0300-8ffa-11eb-8278-bfd6523cd27f.png)

So this seems a browser issue.

I tried to generate many screenshot on virtual machines with BrowserStack (URL = https://www.openstreetmap.org/#map=8/44.056/0.228&layers=Y because the cyclosm website shows the full-page popup on initial load). Unfortunately, there is no Brave browser there. See the results in https://www.browserstack.com/screenshots/7d74e27b6fae0e72746ffb12f4e0bde1e567e157.
